Sources/SourceKitLSP/SourceKitServer.swift

```diff
@@ -1551,25 +1551,24 @@ extension SourceKitServer {
         subsequence: true,
         ignoreCase: true
       ) { symbol in
+        if Task.isCancelled {
+          return false
+        }
         guard !symbol.location.isSystem && !symbol.roles.contains(.accessorOf) else {
           return true
         }
         symbolOccurrenceResults.append(symbol)
-        // FIXME: Once we have cancellation support, we should fetch all results and take the top
-        // `maxWorkspaceSymbolResults` symbols but bail if cancelled.
-        //
-        // Until then, take the first `maxWorkspaceSymbolResults` symbols to limit the impact of
-        // queries which match many symbols.
-        return symbolOccurrenceResults.count < maxWorkspaceSymbolResults
+        return true
       }
+      try Task.checkCancellation()
     }
-    return symbolOccurrenceResults
+    return symbolOccurrenceResults.sorted()
```

Comment:
From looking at indexstore-db this is by location, then roles, then symbol. And the symbol is by usr and then by name.

Maybe that makes sense? The usr then name seems a bit odd to me. It would look pretty random to a user.

Reply:
Honestly, given that it sorts by location first, I don’t really care about the remaining sorting. As long as we get something that’s deterministic from it, I’m fine.